src/Auth.ts

```diff
@@ -199,46 +200,43 @@ export class Auth {
   }
 
   public async ensureAccessToken(resource: string, logger: Logger, debug: boolean = false, fetchNew: boolean = false): Promise<string> {
-    const now: Date = new Date();
-    const accessToken: AccessToken | undefined = this.connection.accessTokens[resource];
-    const expiresOn: Date = accessToken && accessToken.expiresOn ?
-      // if expiresOn is serialized from the service file, it's set as a string
-      // if it's coming from MSAL, it's a Date
-      typeof accessToken.expiresOn === 'string' ? new Date(accessToken.expiresOn) : accessToken.expiresOn
-      : new Date(0);
+    let getTokenPromise: ((resource: string, logger: Logger, debug: boolean, fetchNew: boolean) => Promise<AccessToken | null>) | undefined;
 
-    if (!fetchNew && accessToken && expiresOn > now) {
-      if (debug) {
-        await logger.logToStderr(`Existing access token ${accessToken.accessToken} still valid. Returning...`);
-      }
-      return accessToken.accessToken;
-    }
-    else {
-      if (debug) {
-        if (!accessToken) {
-          await logger.logToStderr(`No token found for resource ${resource}.`);
+    // If the authType is accessToken, we handle returning the token later on.
```

Comment:
if authType is accessToken, shouldn't we just return the access token and be done with it?

Reply:
That's possible, but in that case there's some logic that is skipped, like calling the `storeConnectionInfo` function and setting the identityId parameters etc. That was the reason I wanted to keep things similar to how they are with the other auth methods.

Reply:
I've rewritten it slightly. Better like this? We still should not return it right away.